### PR TITLE
Update CodeEditorView for Horizontal Scroll

### PR DESCRIPTION
**Pull request by Dennis1543**
Updated CodeEditorView to track horizontal offset, align mouse cursor when scrolling, and hide Line Number bar when scrolled.

(I apologize about the format, I'm not quite sure what happened with the commit, but here are the actual changed lines in the new CodeEditorView.xaml.cs:
42-44, 65, 67, 78, 86-102, 284)

For [RATools Issue #303](https://github.com/Jamiras/RATools/issues/303)

### CODE REVIEW

File: UI/WPF/Source/Controls/CodeEditorView.xaml [hunk 1]
```diff
@@ -203,7 +203,8 @@
 
             <!-- line number/line separator -->
             <Border BorderThickness="0,0,1,0" Margin="0,0,2,0" IsHitTestVisible="False" 
-                    BorderBrush="{Binding Resources.LineNumber.Brush}" VerticalAlignment="Stretch" HorizontalAlignment="Right" />
+                    BorderBrush="{Binding Resources.LineNumber.Brush}" VerticalAlignment="Stretch" HorizontalAlignment="Right"
+                    x:Name="LineNum"/>
```
Comment (Jamiras) [inline UI/WPF/Source/Controls/CodeEditorView.xaml:207]: Please use `camelCase` for names.

File: UI/WPF/Source/Controls/CodeEditorView.xaml.cs [hunk 2]
```diff
@@ -1,370 +1,396 @@
...
+            // show / hide the line number border
+            Border linenum = this.FindName("LineNum") as Border;
+            if ( (double)ViewModel.GetValue(HorizontalScrollOffsetProperty) <= 0.1 )
+            {
+                linenum.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                linenum.Visibility = Visibility.Hidden;
+            }
```
Comment (Jamiras) [inline UI/WPF/Source/Controls/CodeEditorView.xaml.cs:100]: The line disappears as soon as I start scrolling horizontally, but the line numbers are still visible:

image
image

Also note that the line appears to go through the scrollbar

Ideally, the line numbers and separator would remain visible and only the content would scroll horizontally.

**Review by Jamiras — COMMENTED** [review]
The formatting changes to CodeEditorViewModel.cs are probably line endings. It looks like you changed the permissions on some of the other files as well. At a minimum, I'd expect those changes to be eliminated to accept this PR.

I've left some feedback, but ultimately decided to make the line numbers not be part of the horizontal scrolling, and took it upon myself to make those changes (#27).

Hopefully this still counts for your contribution credit. For what it's worth, you have my approval. If you could fix the changes so only the modifications showed up and change the one name to camelCase, I would have merged this. It's a functional solution, I just wanted to tweak it a bit.

**Comment by Dennis1543** [comment]
Thanks for the feedback. I'll get credit regardless of whether the PR was accepted, so no worries there. I'm happy you were able to use the functional part of the PR, and this process was quite the learning experience for me. 

Excited to see the changes deployed! Hopefully I'll be able to continue learning, and possibly contribute to more RA tools and applications. I'll work on a better understanding of Git so that next time my PR is much better formatted and legible. Thanks again for the opportunity, and all that you do for RA!

**Comment by Jamiras** [comment]
Closing in favor of #27

**Pull request closed without merging**